thetis/coupled_timeintegrator_2d.py

```diff
@@ -78,21 +82,38 @@ def _create_tracer_integrator(self):
 
         if self.solver.options.solve_tracer:
             uv, elev = self.fields.solution_2d.split()
-            fields = {'elev_2d': elev,
-                      'uv_2d': uv,
-                      'diffusivity_h': self.options.horizontal_diffusivity,
-                      'source': self.options.tracer_source_2d,
-                      'lax_friedrichs_tracer_scaling_factor': self.options.lax_friedrichs_tracer_scaling_factor,
-                      'tracer_advective_velocity_factor': self.options.tracer_advective_velocity_factor,
-                      }
+            fields = {
+                'elev_2d': elev,
+                'uv_2d': uv,
+                'diffusivity_h': self.options.horizontal_diffusivity,
+                'source': self.options.tracer_source_2d,
+                'lax_friedrichs_tracer_scaling_factor': self.options.lax_friedrichs_tracer_scaling_factor,
+                'tracer_advective_velocity_factor': self.options.tracer_advective_velocity_factor,
+            }
             if issubclass(self.tracer_integrator, timeintegrator.CrankNicolson):
-                self.timesteppers.tracer = self.tracer_integrator(solver.eq_tracer, solver.fields.tracer_2d,
-                                                                  fields, solver.dt, bnd_conditions=solver.bnd_functions['tracer'],
-                                                                  solver_parameters=self.options.timestepper_options.solver_parameters_tracer,
-                                                                  semi_implicit=self.options.timestepper_options.use_semi_implicit_linearization,
-                                                                  theta=self.options.timestepper_options.implicitness_theta)
+                self.timesteppers.tracer = self.tracer_integrator(
+                    solver.eq_tracer, solver.fields.tracer_2d, fields, solver.dt,
+                    bnd_conditions=solver.bnd_functions['tracer'],
+                    solver_parameters=self.options.timestepper_options.solver_parameters_tracer,
+                    semi_implicit=self.options.timestepper_options.use_semi_implicit_linearization,
+                    theta=self.options.timestepper_options.implicitness_theta,
+                )
+            elif issubclass(self.tracer_integrator, rungekutta.SSPRK33) or \
```

Comment:
It's somewhat unfortunate that we need these if statements. The idea of a unified timeintegrator API is that one could just directly construct `self.tracer_integrator` object without any checking... Now the optional arguments seem to cause problems. Still, there should be a more succinct way to do this.